storage/api/Storage/Storage.cs

```diff
@@ -407,6 +413,113 @@ private void PrintBucketAclForUser(string bucketName, string userEmail)
+            Console.WriteLine($"Time Created: {metadata.TimeCreated}");
+            Console.WriteLine($"Time Updated: {metadata.Updated}");
+            Console.WriteLine($"ETag: {metadata.ETag}");
+        }
+        // [END storage_get_hmac_key]
+
+        // [START storage_deactivate_hmac_key]
+        private void DeactivateHmacKey(String accessId)
+        {
+            var storage = StorageClient.Create();
+            var metadata = storage.GetHmacKey(s_projectId, accessId);
+            metadata.State = HmacKeyStates.Inactive;
+            var updatedMetadata = storage.UpdateHmacKey(metadata);
+
+            Console.WriteLine("The HMAC key is now inactive.");
+            Console.WriteLine("The HMAC key metadata is:");
+            Console.WriteLine($"ID: {updatedMetadata.Id}");
+            Console.WriteLine($"Access ID: {updatedMetadata.AccessId}");
+            Console.WriteLine($"Project ID: {updatedMetadata.ProjectId}");
+            Console.WriteLine($"Service Account Email: {updatedMetadata.ServiceAccountEmail}");
+            Console.WriteLine($"State: {updatedMetadata.State}");
+            Console.WriteLine($"Time Created: {updatedMetadata.TimeCreated}");
+            Console.WriteLine($"Time Updated: {updatedMetadata.Updated}");
+            Console.WriteLine($"ETag: {updatedMetadata.ETag}");
+        }
+        // [END storage_deactivate_hmac_key]
+
+        // [START storage_activate_hmac_key]
+        private void ActivateHmacKey(String accessId)
+        {
+            var storage = StorageClient.Create();
+            var metadata = storage.GetHmacKey(s_projectId, accessId);
+            metadata.State = HmacKeyStates.Active;
+            var updatedMetadata = storage.UpdateHmacKey(metadata);
+
+            Console.WriteLine("The HMAC key is now active.");
+            Console.WriteLine("The HMAC key metadata is:");
+            Console.WriteLine($"ID: {updatedMetadata.Id}");
+            Console.WriteLine($"Access ID: {updatedMetadata.AccessId}");
+            Console.WriteLine($"Project ID: {updatedMetadata.ProjectId}");
+            Console.WriteLine($"Service Account Email: {updatedMetadata.ServiceAccountEmail}");
+            Console.WriteLine($"State: {updatedMetadata.State}");
+            Console.WriteLine($"Time Created: {updatedMetadata.TimeCreated}");
+            Console.WriteLine($"Time Updated: {updatedMetadata.Updated}");
+            Console.WriteLine($"ETag: {updatedMetadata.ETag}");
+        }
+        // [END storage_activate_hmac_key]
+
+        // [START storage_delete_hmac_key]
+        private void DeleteHmacKey(String accessId)
+        {
+            var storage = StorageClient.Create();
+            storage.DeleteHmacKey(s_projectId, accessId);
+
+            Console.WriteLine($"Key {accessId} was deleted.");
+        }
+        // [END storage_delete_hmac_key]
+
+        // [START storage_list_hmac_keys]
+        private void ListHmacKeys(String serviceAccountEmail = null)

```

Comment:
Let's keep it consistent with other languages where we have not filtered the listing with `serviceAccountEmail`.

Reply:
Thanks!